packages/shared/lib/core/network/utils/ensureSinglePrimaryNode.ts

```diff
@@ -0,0 +1,23 @@
+import { INode } from '../interfaces'
+import { setRandomPrimaryNode } from './setRandomPrimaryNode'
+
+/**
+ * Ensures that a list of nodes contains only one primary node. If none exist, one will
+ * be selected randomly.
+ * @method ensureSinglePrimaryNode
+ * @param {INode[]} nodes
+ * @returns {INode[]}
+ */
+export function ensureSinglePrimaryNode(nodes: INode[]): INode[] {
+    if (!nodes || !nodes.length) return nodes
```

Comment:
Probably would be better to return an empty array `[]`, esp. if `nodes` is `null` or `undefined` for whatever reason, rather than return `nodes`.

Reply:
Just copied this from what it was previously, can make this change though.

Reply:
Done